nornir/core/__init__.py

```diff
@@ -41,32 +14,30 @@ class Nornir(object):
 
     Arguments:
         inventory (:obj:`nornir.core.inventory.Inventory`): Inventory to work with
-        data(:obj:`nornir.core.Data`): shared data amongst different iterations of nornir
+        data(GlobalState): shared data amongst different iterations of nornir
         dry_run(``bool``): Whether if we are testing the changes or not
         config (:obj:`nornir.core.configuration.Config`): Configuration object
         config_file (``str``): Path to Yaml configuration file
 
     Attributes:
         inventory (:obj:`nornir.core.inventory.Inventory`): Inventory to work with
-        data(:obj:`nornir.core.Data`): shared data amongst different iterations of nornir
+        data(:obj:`nornir.core.GlobalState`): shared data amongst different iterations of nornir
         dry_run(``bool``): Whether if we are testing the changes or not
         config (:obj:`nornir.core.configuration.Config`): Configuration parameters
     """
 
     def __init__(
-        self, inventory, dry_run, config=None, config_file=None, logger=None, data=None
+        self, inventory, _config=None, config_file=None, logger=None, data=None
```

Comment:
That's because config is a property and there is some magic with the `filter` method. However, this can be fixed in #252, I just don't want to overcomplicate this PR even more.